src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java

```diff
@@ -600,29 +604,56 @@ public Response showSettings(@Context HttpServletRequest request) throws IOExcep
 	@POST
 	@Path("/branches/delete")
 	@Transactional
-	public Response deleteBehindBranch(@Context HttpServletRequest request,
-									   @FormParam("branchDeleteName") String branchDeleteName,
-									   @QueryParam("fatal") String fatal)
+	public Response deleteBranch(@Context HttpServletRequest request,
+                                 @FormParam("branchDeleteName") String branchDeleteName,
```

Comment:
hmm no that won't work, as it is a `POST` request and not a `GET`. (We could change that as well.) But I wouldn't change it just now, we have bigger issues to solve :wink: Have a look at replacing the "magic" strings with an enum though. JAX-RS is able to parse Enum values from strings, so it should be easy. And then you can make the logic less verbose/error prone/magic  and move some of the documentation to the enum  values instead .